Homework/Homework 1_WM.ipynb

```diff
@@ -0,0 +1,1736 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "A_WM=[[0,0,0,0,0,0,0,0,1,0],#0\n",
+    "            [0,0,1,0,1,0,0,0,0,0],#1\n",
+    "            [0,1,0,1,0,1,0,0,0,0],#2\n",
+    "            [0,0,1,0,0,0,1,0,0,0],#3\n",
+    "            [0,1,0,0,0,1,0,1,0,0],#4\n",
+    "            [0,0,1,0,1,0,1,0,1,0],#5\n",
+    "            [0,0,0,1,0,1,0,0,0,1],#6\n",
+    "            [0,0,0,0,1,0,0,0,1,0],#7\n",
+    "            [1,0,0,0,0,1,0,1,0,1],#8\n",
+    "            [0,0,0,0,0,0,1,0,1,0]]#9\n",
+    "    \n",
+    "O_WM=[[1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1],\n",
+    "            [1]]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def MatrixMultiplication(A,B):\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    A function which computes matrix multiplication of the first and second inputs. Inputs should be list-like.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    if len(A[0]) ==len(B):\n",
+    "        R=[[sum(x*y for x,y in zip(A_row,B_col)) for B_col in zip(*B)]\n",
+    "           for A_row in A]\n",
+    "        return R\n",
+    "    else:\n",
+    "             print(\"Matrix dimensions do not match\")\n",
+    "\n",
+    "\n",
+    "class Matrix_Set:\n",
+    "    def __init__(self,A,O,S,N):\n",
+    "        self.A=A\n",
+    "        self.A_1 = []\n",
+    "        self.A_2 = []\n",
+    "        self.A_3 = []\n",
+    "        self.S = S\n",
+    "        self.O = O\n",
+    "        self.N = N\n",
+    "        self.t1 = 0\n",
+    "        self.t2 = 0\n",
+    "        self.time_elapsed = []\n",
+    "        self.N_used = []\n",
+    "    \n",
+    "\n",
+    "    \n",
+    "    def Dialer_Problem_2(self):\n",
+    "         # same function as problem c without numpy \n",
+    "            \n",
+    "    \n",
+    "        self.t1 = time.time()\n",
+    "        C = MatrixMultiplication(self.A,self.O)\n",
+    "        if(self.S<0 or self.S>9 or self.N <= 0): \n",
+    "            return 0\n",
+    "        if(self.N == 1): \n",
+    "            return 1\n",
+    "        for i in range(self.N-2):\n",
+    "            C = MatrixMultiplication(A,C)\n",
+    "            \n",
+    "        self.t2 = time.time()\n",
+    "            #print('It took : '+str(t2-t1)+ ' seconds')\n",
+    "        print ('It took {s} seconds'.format(s=self.t2-self.t1))\n",
+    "        return(C[self.S])\n",
+    "    \n",
+    "    \n",
+    "    def Performance(self):\n",
+    "        from matplotlib import pyplot as plt\n",
```

Comment:
It is better to import libraries outside the function at the beginning of the code.